### PR TITLE
Preventing modals to close when highlighting text and releasing outside fix#152

### PR DESCRIPTION
**Pull request by deep34**

### CODE REVIEW

**Comment by adamkdean** [comment]
https://github.com/edge/wallet/issues/152

File: src/components/Modal.vue [hunk 1]
```diff
@@ -41,15 +41,33 @@
...
+      }
+    },
+    mounted()
+    {
+        window.addEventListener("mousedown", this.onMouseDown);
+    },
+    beforeUnmout()
+    {
+        window.removeEventListener("mousedown", this.onMouseDown);
+    },
```
Comment (annybs) [inline src/components/Modal.vue:56]: The Modal component is not responsive to mounting/unmounting events when it is shown/hidden. Modal instances are rendered into the page irrespective of visibility, which is controlled by the `visible` prop.

As such, these event listeners should be toggled by watching the `visible` value instead. Please see [SendModal](https://github.com/edge/wallet/blob/b2cabec879172414ef9721740d40f0d7fa2beb0b/src/components/tx/SendModal.vue#L247-L254) for an example.

(Another, minor point that won't matter after the above change: [beforeUnmount](https://v3.vuejs.org/api/options-lifecycle-hooks.html#beforeunmount) is misspelled)

File: src/components/Modal.vue [hunk 2]
```diff
@@ -41,15 +41,33 @@
...
         }
       },
       onClickOutside() {
-        if (this.preventClickOutside) return
+        if (this.preventClickOutside || this.mouseDownInsideModal) return
         if (this.close) this.close()
+      },
+      onMouseDown(mouseEvent)
+      {
+          const modalOuter = document.querySelector(".modal-outer");
```
Comment (annybs) [inline src/components/Modal.vue:69]: Please use [a ref](https://v3.vuejs.org/guide/component-template-refs.html) instead of DOM API.

**Review by annybs — CHANGES_REQUESTED** [review]
The basic logic is sound, but a few changes are needed before we can merge this work.

Please note in addition to the code comments, and with reference to the new [Contributor Guidelines](https://github.com/edge/wallet/blob/develop/CONTRIBUTING.md):

- In this project we do not use semicolons at the end of lines. These should be removed for consistency
- The PR must be rebased onto `develop` branch and the target branch changed to `develop` also

Thanks again for your contribution.

**Comment by annybs** [comment]
@deep34 Please see #159 reflecting a scope change.

After an internal discussion about UX, we are likely to remove the behaviour of closing modals when clicking outside them.

You are still welcome to complete this PR by resolving my feedback, as there isn't a firm timeline for this change. However, I think it's likely to happen within the next 1 week, so it may not be worth your time to continue working on this. If you agree with that, you can close your PR, otherwise we'll close it for you after that change is implemented.

Thanks again for your contribution either way!

**Comment by deep34** [comment]
@annybs No problem, i will close the PR and search for other issues to fix.

**Pull request closed without merging**